src/circuit/gadgets/constraints.rs

```diff
@@ -99,6 +120,61 @@ pub(crate) fn popcount<F: PrimeField, CS: ConstraintSystem<F>>(
     Ok(())
 }
 
+/// Adds a constraint to CS, enforcing that the addition of the allocated numbers in vector `v`
+/// is equal to `one`.
+///
+/// summation(v) = one
+#[allow(dead_code)]
+pub(crate) fn popcount_one<F: PrimeField, CS: ConstraintSystem<F>>(
+    cs: &mut CS,
+    v: &[Boolean],
+) -> Result<(), SynthesisError> {
+    let mut v_lc = LinearCombination::<F>::zero();
+    for b in v {
+        v_lc = add_to_lc::<F, CS>(b, v_lc, F::one())?;
+    }
```

Comment:
You can use `fold` here, too. It's equivalent but makes the meaning a little clearer, in my opinion.

Reply:
Is there a monadic `fold` that allows the use of `?` in the accumulation function?

Reply:
`try_fold`

Reply:
Addressed in #404